.github/workflows/ci.yml

```diff
@@ -0,0 +1,46 @@
+name: CI
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+
+env:
+  # The NAME makes it easier to copy/paste snippets from other CI configs
+  NAME: mir-json
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Deps
+      run: |
+        rustup default nightly-2023-01-23
+        rustup component add clippy rustc-dev
+    - uses: Swatinem/rust-cache@v2.7.0
+    # TODO: Fix lints, use --deny warnings
+    - name: Lint
+      run: cargo clippy
+
+  test:
```

Comment:
Perhaps this should be named "`build`" instead of "`test`"? At present, this job doesn't run any tests.